### ISSUE TITLE
PlanetScale incorrect syncing

### ISSUE THREAD

**Issue by omaralsoudanii** [leerob/leerob.io#348]
Hi @leerob 

I noticed in some of the pull requests created by other contributors a wired behavior with (Vercel / PlanetScale / Your repo).

The behavior is that the preview URL generated by Vercel **have the same numbers and data pulled from PlanetScale as your main domain**.

For example this [pull request](https://github.com/leerob/leerob.io/pull/346#issuecomment-881971463) generated a preview URL.

Open the [preview URL](https://leerob-git-fork-deep-codes-copy-clip-leerob1.vercel.app) and your [main domain](https://leerob.io/) on 2 tabs. 
From the preview URL visit any blog post and keep visiting the same post with back/forward (to increase the views and re-produce the issue).

Then from your main domain tab go to blog posts page or the same blog you used on preview URL (doesn't matter much, in fact with swr when I do this and see the other tab it auto sync) and you will see they have the same views. The preview URL views increase syncs it to your main domain. 

Not sure how you handle PlanetScale branching stuff (or maybe it's automatic on vercel?) but to me it seems both are working on the production branch of the database. Since it's a database issue then I would believe the same apply to any functionality using the database instance.  

The weird thing is that if the pull request created by you then I'm not seeing this behavior! which lead me to believe the root cause only forked repositories from another users causing this. (also what happens if the pull request is merged? didn't check this)

You can see the same thing on guestbook page [here](https://leerob.io/guestbook) and [here](https://leerob-pwp1x6cp7-leerob1.vercel.app/guestbook). I will delete the message after your review.

Personally I just solved this using a new environment variable with the name of the PlanetScale branch I want to connect to (how does it work with you locally since you connect to `main`?) Like below.

`.env`  or `.env.local`
```env
PLANETSCALE_BRANCH=my_local_development_branch_name
```

Your PlanetScale file that connects to an instance `planetscale.js`
```js
const db = new PSDB(process.env.PLANETSCALE_BRANCH)
```

Basically a value to the branch I want to connect to through environment variable.

**Comment by leerob** [comment]
For testing, I was using the same database from Preview and Production, so both environment variables were pointed to the same place. I just went ahead and switched Preview to use a different PlanetScale database.

Eventually, I will hook this up so it uses branch databases inside PlanetScale rather than a separate set of environment variables 👍
